### PR TITLE
[lldb] Update swift::ASTContext API

### PR DESCRIPTION
**Pull request by cachemeifyoucan**
Update lldb to use updated swift::ASTContext API.

### CODE REVIEW

**Comment by cachemeifyoucan** [comment]
Please test with following pull request:
https://github.com/apple/swift/pull/71497

@swift-ci please test

**Comment by cachemeifyoucan** [comment]
Please test with following pull request:
https://github.com/apple/swift/pull/71497

@swift-ci please test

**Comment by cachemeifyoucan** [comment]
Please test with following pull request:
https://github.com/apple/swift/pull/71497

@swift-ci please test windows platform

**Review by adrian-prantl — APPROVED** [review]

File: lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp [hunk 1]
```diff
@@ -3104,6 +3104,10 @@ swift::symbolgraphgen::SymbolGraphOptions &SwiftASTContext::GetSymbolGraphOption
   return GetCompilerInvocation().getSymbolGraphOptions();
 }
 
+swift::CASOptions &SwiftASTContext::GetCASOptions() {
+  return GetCompilerInvocation().getCASOptions();
```
Comment (adrian-prantl) [inline lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp:3108]: Do the CAS options get serialized in the Swift module together with the rest of the invocation?

Comment (cachemeifyoucan) [reply lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp:3108]: I actually have a follow up patch to include that in the context hash. Does debugger need something from CASOption to be serialized into module?

**Pull request merged**